zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/authorization/DbAuthorizationState.java

```diff
@@ -21,65 +22,84 @@
-    ownerType = new DbString();
-    resourceKey = new DbString();
+    ownerKey = new DbLong();
     resourceType = new DbString();
-    resourceCompositeKey = new DbCompositeKey<>(resourceKey, resourceType);
-    ownerAndResourceCompositeKey =
-        new DbCompositeKey<>(new DbCompositeKey<>(ownerKey, ownerType), resourceCompositeKey);
+    permission = new DbString();
+    resourceTypeAndPermissionCompositeKey = new DbCompositeKey<>(resourceType, permission);
+    ownerKeyAndResourceTypeAndPermissionCompositeKey =
+        new DbCompositeKey<>(ownerKey, resourceTypeAndPermissionCompositeKey);
 
-    ownerAuthorizationColumnFamily =
+    resourceIdsByOwnerKeyResourceTypeAndPermissionColumnFamily =
         zeebeDb.createColumnFamily(
-            ZbColumnFamilies.AUTHORIZATIONS_BY_USERNAME_AND_PERMISSION,
+            ZbColumnFamilies.RESOURCE_IDS_BY_OWNER_KEY_RESOURCE_TYPE_AND_PERMISSION,
             transactionContext,
-            ownerAndResourceCompositeKey,
-            persistedPermissions);
+            ownerKeyAndResourceTypeAndPermissionCompositeKey,
+            resourceIdentifiers);
+
+    resourceId = new DbString();
+    resourceIdAndOwnerKeyCompositeKey = new DbCompositeKey<>(resourceId, ownerKey);
+    authorizationKeyByResourceIdAndOwnerKeyColumnFamily =
+        zeebeDb.createColumnFamily(
+            ZbColumnFamilies.AUTHORIZATION_KEY_BY_RESOURCE_ID_AND_OWNER_KEY,
+            transactionContext,
+            resourceIdAndOwnerKeyCompositeKey,
+            ownerKeyAndResourceTypeAndPermissionCompositeKey);
+
+    ownerType = new DbString();
+    ownerTypeByOwnerKeyColumnFamily =
+        zeebeDb.createColumnFamily(
+            ZbColumnFamilies.OWNER_TYPE_BY_OWNER_KEY, transactionContext, ownerKey, ownerType);
   }
 
   @Override
   public void createAuthorization(final AuthorizationRecord authorizationRecord) {
     persistedAuthorization.setAuthorization(authorizationRecord);
 
-    ownerKey.wrapString(authorizationRecord.getOwnerKey());
+    ownerKey.wrapLong(authorizationRecord.getOwnerKey());
     ownerType.wrapString(authorizationRecord.getOwnerType().name());
-    resourceKey.wrapString(authorizationRecord.getResourceKey());
     resourceType.wrapString(authorizationRecord.getResourceType());
-    persistedPermissions.setPermissions(authorizationRecord.getPermissions());
-    ownerAuthorizationColumnFamily.insert(ownerAndResourceCompositeKey, persistedPermissions);
+    resourceIdentifiers.setResourceIdentifiers(authorizationRecord.getPermissions());
+    resourceIdsByOwnerKeyResourceTypeAndPermissionColumnFamily.insert(
+        ownerKeyAndResourceTypeAndPermissionCompositeKey, resourceIdentifiers);
   }
 
   @Override
-  public PersistedPermissions getPermissions(
-      final String ownerKey,
-      final AuthorizationOwnerType ownerType,
-      final String resourceKey,
-      final String resourceType) {
-    this.ownerKey.wrapString(ownerKey);
+  public ResourceIdentifiers getPermissions(
```

Comment:
❓ I think the naming here can be changed I guess its no longer `getPermissions` but more `getResourceIdentifiers`

Reply:
Good catch